models/unet/train_unet.py

```diff
@@ -90,248 +87,138 @@ def __call__(self, kspace, target, attrs, fname, slice):
+            'slice': slice,
+            'output': (output * std + mean).cpu().numpy(),
+            'target': (target * std + mean).cpu().numpy(),
+            'val_loss': F.l1_loss(output, target),
+        }
+
+    def test_step(self, batch, batch_idx):
+        input, _, mean, std, fname, slice = batch
+        output = self.forward(input)
+        mean = mean.unsqueeze(1).unsqueeze(2)
+        std = std.unsqueeze(1).unsqueeze(2)
+        return {
+            'fname': fname,
+            'slice': slice,
+            'output': (output * std + mean).cpu().numpy(),
+        }
+
+    def configure_optimizers(self):
+        optim = RMSprop(self.parameters(), lr=self.hparams.lr, weight_decay=self.hparams.weight_decay)
+        scheduler = torch.optim.lr_scheduler.StepLR(optim, self.hparams.lr_step_size, self.hparams.lr_gamma)
+        return [optim], [scheduler]
+
+    def train_data_transform(self):
+        mask = create_mask_for_mask_type(self.hparams.mask_type, self.hparams.center_fractions,
+                                         self.hparams.accelerations)
+        return DataTransform(self.hparams.resolution, self.hparams.challenge, mask, use_seed=False)
+
+    def val_data_transform(self):
+        mask = create_mask_for_mask_type(self.hparams.mask_type, self.hparams.center_fractions,
+                                         self.hparams.accelerations)
+        return DataTransform(self.hparams.resolution, self.hparams.challenge, mask)
+
+    def test_data_transform(self):
+        return DataTransform(self.hparams.resolution, self.hparams.challenge)
+
+    @staticmethod
+    def add_model_specific_args(parser):
+        parser.add_argument('--num-pools', type=int, default=4, help='Number of U-Net pooling layers')
+        parser.add_argument('--drop-prob', type=float, default=0.0, help='Dropout probability')
+        parser.add_argument('--num-chans', type=int, default=32, help='Number of U-Net channels')
+        parser.add_argument('--batch-size', default=16, type=int, help='Mini batch size')
+        parser.add_argument('--lr', type=float, default=0.001, help='Learning rate')
+        parser.add_argument('--lr-step-size', type=int, default=40,
+                            help='Period of learning rate decay')
+        parser.add_argument('--lr-gamma', type=float, default=0.1,
+                            help='Multiplicative factor of learning rate decay')
+        parser.add_argument('--weight-decay', type=float, default=0.,
+                            help='Strength of weight decay regularization')
+        return parser
 
 
 def main(args):
-    args.exp_dir.mkdir(parents=True, exist_ok=True)
-    writer = SummaryWriter(log_dir=args.exp_dir / 'summary')
-
-    if args.resume:
-        checkpoint, model, optimizer = load_model(args.checkpoint)
-        args = checkpoint['args']
-        best_dev_loss = checkpoint['best_dev_loss']
-        start_epoch = checkpoint['epoch']
-        del checkpoint
+    if args.mode == 'test':
```

Comment:
Maybe debatable but imo this function would be slightly easier to read with a local get_trainer_for_logger function and then one if statement controlling the test and train flow.